### PR TITLE
support lock table

### PR DESCRIPTION
**Pull request by marsishandsome**
close https://github.com/pingcap/tispark/issues/818

### CODE REVIEW

**Comment by marsishandsome** [comment]
# tidb with table lock test success
https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tispark_ghpr_integration_test/detail/tispark_ghpr_integration_test/558/pipeline/98
image

# tidb without table lock test success (skip)
https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tispark_ghpr_integration_test/detail/tispark_ghpr_integration_test/560/pipeline
image

**Comment by marsishandsome** [comment]
@zhexuany @birdstorm PTAL

File: core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala [hunk 1]
```diff
@@ -359,6 +401,41 @@ class TiBatchWrite(@transient val df: DataFrame,
...
+  private def lockTable(): Unit = {
+    if (isEnableTableLock) {
+      if (!tableLocked) {
+        tiDBJDBCClient.lockTableWriteLocal(options.database, options.table)
+        tableLocked = true
+      } else {
+        logger.warn("table already locked!")
+      }
+    } else {
+      // TODO: what if version of tidb does not support lock table
```
Comment (zhexuany) [inline core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala]: In this case, just throw an exception and abort this operation.

File: core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala [hunk 2]
```diff
@@ -359,6 +401,41 @@ class TiBatchWrite(@transient val df: DataFrame,
...
+  private def unlockTable(): Unit = {
+    if (isEnableTableLock) {
+      if (tableLocked) {
+        tiDBJDBCClient.unlockTables()
+        tableLocked = false
+      } else {
+        logger.warn("table already unlocked!")
+      }
+    } else {
+      // TODO: what if version of tidb does not support lock table
```
Comment (zhexuany) [inline core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala]: ditto

File: core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala (hunk 1 above)
Comment (marsishandsome) [reply core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala]: i will do it in this issue

https://github.com/pingcap/tispark/issues/839

**Pull request merged**